src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1420,25 +1384,34 @@ void LedgerConsensusImp::statusChange (
     WriteLog (lsTRACE, LedgerConsensus) << "send status change to peer";
 }
 
-void LedgerConsensusImp::takeInitialPosition (Ledger& initialLedger)
+void LedgerConsensusImp::takeInitialPosition (
+    std::shared_ptr<ReadView const> const& initialLedger)
 {
-    std::shared_ptr<SHAMap> initialSet;
+    std::shared_ptr<SHAMap> initialSet = std::make_shared <SHAMap> (
+        SHAMapType::TRANSACTION, getApp().family(), deprecatedLogs().journal("SHAMap"));
+
+    // Build SHAMap containing all transactions in our open ledger
+    for (auto const& tx : initialLedger->txs)
+    {
+        Serializer s (2048);
+        tx.first->add(s);
+        initialSet->addItem (
+            SHAMapItem (tx.first->getTransactionID(), s), true, false);
```

Comment:
I just reviewed @HowardHinnant's PR that adds move support for `Serializer` in `SHAMapItem`. Moving from `s` here will save one memory allocation and copy per loop iteration (after his PR is merged).